Makefile

```
@@ -66,12 +89,14 @@ metadata: lib all
 
 test: all
 	$(MAKE) -C haskell test PACKAGE=segmented-cfmm \
```

We can do the same thing here:
```suggestion
test: prepare_lib
	$(MAKE) -C haskell test PACKAGE=segmented-cfmm
```

---

Updated.